### ISSUE TITLE
Allow mapping via fluent API

### ISSUE THREAD

**Issue by cremor** [efcore/EFCore.CheckConstraints#165]
It would be great if this library could also provide fluent mapping APIs to allow mapping of check constraints without using attributes.

Example:
There is no `HasMinLength()` fluent mapping method. So you have to use a `[StringLength(2, MinimumLength = 2)]` attribute to get a min length check constraints.

**Comment by roji** [comment]
@cremor the point of this plugin is to provide support for standard .NET plugins which are already there - [StringLength] is part of .NET. If you just want to specify a minimum length check constraint in EF via the Fluent API, you can simply define the check constraint directly via EF's API ([docs](https://learn.microsoft.com/en-us/ef/core/modeling/indexes?tabs=data-annotations#check-constraints)).

**Comment by cremor** [comment]
Yes, but with manually written check constraints I also have to manually do all the work that this library is already doing when I use attributes, like:
- building the constraint name
- using the correct column name
- correctly quoting the column name

I like that this library does all that automatically 😄

**Comment by roji** [comment]
I'll leave this open in the backlog, but all the above really is trivial stuff - this plugin really was meant for making the built-in .NET attributes work, not making a really simple task even a big simpler :)

**Comment by martin-hirsch** [comment]
I am also configuring EF Core with Fluent API and hoped to find a way to gradually add Check Constraints for Enums.
As far as I understand it, there is only an option to turn them on for all Enums? `.UseEnumCheckConstraints();` is not configurable.

I think it would be helpful for non-greenfield projects to be able to turn this on for specific Enums. What do you think?

**Comment by roji** [comment]
@martin-hirsch the point is that if you want to have check constraints only for specific enums, you're going to have to go and configure them on an enum-by-enum basis, and at that point it's really very trivial to just configure the check constraint yourself. In other words, the plugin doesn't provide that much value if you're doing that.

**Comment by martin-hirsch** [comment]
@roji, I agree. The term _plugin_ convinced me. :-)

I just thought it would be nice to grab this fine pice of NuGet and don't have to invent my own interpretation of a generic way to constrain specific Enums in my solutions. Work my way up until eventually I can flip the switch.

Thanks for your quick response!

**Comment by NickStrupat** [comment]
@roji would you be open to merging a PR that implemented this?
